shell/platform/darwin/ios/rendering_api_selection.mm

```diff
@@ -0,0 +1,78 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/rendering_api_selection.h"
+
+#include <Foundation/Foundation.h>
+#include <QuartzCore/CAEAGLLayer.h>
+#include <QuartzCore/CAMetalLayer.h>
+
+#include "flutter/fml/logging.h"
+
+namespace flutter {
+
+bool ShouldUseSoftwareRenderer() {
+  return [[[NSProcessInfo processInfo] arguments] containsObject:@"--force-software"];
+}
+
+bool ShouldUseMetalRenderer() {
+  // If there is a command line argument that says Metal should not be used, that takes precedence
+  // over everything else. This allows disabling Metal on a per run basis to check for regressions
+  // on an application that has otherwise opted into Metal on an iOS version that supports it.
+  if ([[[NSProcessInfo processInfo] arguments] containsObject:@"--disable-metal"]) {
+    return false;
+  }
+
+  // If the application wants to use metal on a per run basis with disregard for version checks or
+  // plist based opt ins, respect that opinion. This allows selectively testing features on older
+  // version of iOS than those explicitly stated as being supported.
+  if ([[[NSProcessInfo processInfo] arguments] containsObject:@"--force-metal"]) {
+    return true;
+  }
+
+  // This is just a version we picked that is easy to support and has all necessary Metal features.
+  bool ios_version_supports_metal = false;
+  if (@available(iOS 11.0, *)) {
+    ios_version_supports_metal = true;
+  }
+
```

Comment:
Should there be a query of the underlying hardware?  If you have iOS 11 does that mean your phone has to support it?  We should add that info in the comment if so.

Reply:
This is a Skia constraint.  Metal is supported from something like 8.0, but Skia only supports down to 11.0.

Is there some way we could parameterize the Skia constraint here from some constant in Skia?

Reply:
I believe we can go lower still but I haven't tested it yet to confirm. This constant is still WIP and this code was just moved here and not authored anew. I will tackle this separately.

Reply:
can you link in an issue so we don't lose it?

Reply:
Filed https://github.com/flutter/flutter/issues/52356 and also linked in a TODO here.